### PR TITLE
Added IP address validation for source IP of LidarUdpPublisher

### PR DESCRIPTION
**Pull request by hosokawa-ikuto**
I added an IP address validation for LidarUdpPublisher

### CODE REVIEW

**Review by msz-rai — CHANGES_REQUESTED** [review]
Good job! :+1: 
Please consider my suggestions :bowing_man:

File: Assets/RGLUnityPlugin/Scripts/LidarUdpPublisher.cs [hunk 1]
```diff
@@ -268,5 +272,29 @@ private bool IsVelodyne(LidarModel model)
...
+
+        private bool IsValidIpAddress(in string ip)
+        {
+            bool foundMyIp = false;
+            if (ip == "0.0.0.0") return true;
+            foreach (var nic in NetworkInterface.GetAllNetworkInterfaces())
+            {
+                foreach (var unicast in nic.GetIPProperties().UnicastAddresses)
+                {
+                    //if (unicast.Address.AddressFamily == AddressFamily.InterNetworkV6) continue; // skip for IPv6
```
[the remark this answers is not in the document]
Comment (hosokawa-ikuto) [reply Assets/RGLUnityPlugin/Scripts/LidarUdpPublisher.cs:290]:
```suggestion
```

File: Assets/RGLUnityPlugin/Scripts/LidarUdpPublisher.cs [hunk 2]
```diff
@@ -15,6 +15,8 @@
 using UnityEngine;
 using System.Collections.Generic;
 using System.Linq;
+using System.Net.NetworkInformation;
+//using System.Net.Sockets; // for IPv6 checking
```
Comment (hosokawa-ikuto) [inline Assets/RGLUnityPlugin/Scripts/LidarUdpPublisher.cs:19]:
```suggestion
```

**Review by hosokawa-ikuto — COMMENTED** [review]
IPv6 related comments were removed

**Review by mackierx111 — APPROVED** [review]

**Pull request merged**